src/App.tsx

```diff
@@ -1,41 +1,81 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getPosts } from './api/posts';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [userId, setUserId] = useState(0);
+  const [prevUserId, setPrevUserId] = useState(0);
+
+  const changeOpenPost = (value: number) => {
+    setSelectedPostId(value);
+  };
+
+  const getAllPosts = async () => {
+    setPosts(await getPosts(userId));
+  };
+
+  useEffect(() => {
+    getAllPosts();
+  }, []);
+
+  useEffect(() => {
+    if (userId !== prevUserId) {
+      getAllPosts();
+      setPrevUserId(userId);
+    }
+  });
```

Comment:
```suggestion
  useEffect(() => {
      getAllPosts();
    };
  }, [userId]);
```
Встанови масив залежності і тоді не треба буде в стейті зберігати попереднє ід і робити зайву перевірку